Core/src/org/sleuthkit/autopsy/directorytree/DirectoryTreeTopComponent.java

```diff
@@ -201,30 +178,28 @@ public void popupMenuCanceled(PopupMenuEvent e) {
      */
     private void preExpandNodes(Children rootChildren) {
         BeanTreeView tree = getTree();
-        for (String categoryKey : new String[]{AnalysisResults.getName(), DataArtifacts.getName()}) {
-            Node categoryNode = rootChildren.findChild(categoryKey);
-
-            if (!Objects.isNull(categoryNode)) {
-                tree.expandNode(categoryNode);
-                Children resultsChildren = categoryNode.getChildren();
-                Arrays.stream(resultsChildren.getNodes()).forEach(tree::expandNode);
-            }
-        }
 
-        Node views = rootChildren.findChild(ViewsNode.NAME);
-        if (!Objects.isNull(views)) {
-            tree.expandNode(views);
+        // using getNodes(true) to fetch children so that async nodes are loaded
+        Node[] rootChildrenNodes = rootChildren.getNodes(true);
+        if (rootChildrenNodes == null || rootChildrenNodes.length < 1) {
+            return;
         }
 
         // expand all nodes parents of and including hosts if group by host/person
         if (Objects.equals(CasePreferences.getGroupItemsInTreeByDataSource(), true)) {
-            Node[] rootNodes = rootChildren.getNodes();
-            if (rootNodes != null) {
-                Stream.of(rootNodes)
-                        .flatMap((n) -> getHostNodesAndParents(n).stream())
-                        .filter((n) -> n != null)
-                        .forEach((n) -> tree.expandNode(n));
-            }
+            Stream.of(rootChildrenNodes)
+                    .flatMap((n) -> getHostNodesAndParents(n).stream())
+                    .filter((n) -> n != null)
+                    .forEach(tree::expandNode);
+        } else {
+            // nodes to be opened if present at top level
+            Set<String> resultArtifactNames
```

Comment:
Is there a particular reason you are using a set instead of a list?  List are more efficent when adding items.  This method is called on the EDT so we need to be thoughtful about performance.

Reply:
I was trying to optimize for the `contains` operation on line 201 for O(1) performance.

Reply:
If it doesn't need to be created more than once I would move it, however static class fields for DirectoryTreeTopComponent will be constructed on the EDT.

Reply:
Ok.  I just moved it to a static field.